File: src/2018/Day_01/P1.py
def output (data):
    print(f"Result: {data}");
    with open("output.txt", "w") as f:
        f.write(str(data));

def solve (data):
    total = 0;
    for i in range(0, len(data), 1):
        if (data[i][0] == "+"):
            amt = int(data[i][1:]);
        else:
            amt = -1 * int(data[i][1:]);

        total += amt;
    output(total);

with open("input.txt", "r") as f:
    solve(f.readlines());

File: src/2022/Day_04/P1.js
import * as path from 'path';
import * as fs from 'fs';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const data = fs.readFileSync(path.resolve(__dirname, `./input.txt`), `utf-8`);

const expand = range => {
    const coords = [];

    const [x1, x2] = range.split(`-`).map(x => parseInt(x));
    for (let i = x1 - 1; i <= x2 + 1; i++) coords.push(i);

    return coords.map(x => String(x));
};

const main = async () => {
    const lines = data.split(`\n`).map(x => x.trim());
    let ans = 0;

    for (const line of lines) {
        const elves = line.split(`,`);
        const a = expand(elves[0]).join(``);
        const b = expand(elves[1]).join(``);

        if (a.includes(b) || b.includes(a)) ans++;
    }

    fs.writeFileSync(path.resolve(__dirname, `./output.txt`), String(ans));

    if (process.argv.length <= 3) console.log(`Result: ${ans}`);
    return ans;
};

if (process.argv.length <= 3) void main();
export default main;
